powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/ApplicationService.java

```diff
@@ -50,16 +53,39 @@ public ApplicationService(TestConfigRepository appConfigRepository) {
      * Configure an application.
      * @param request Configure an application request.
      * @return Configure an application response.
+     * @throws AppConfigInvalidException Thrown in case mobile SDK configuration is invalid.
      */
     @Transactional
-    public Response configureApplication(ConfigureApplicationRequest request) {
+    public Response configureApplication(final ConfigureApplicationRequest request) throws AppConfigInvalidException {
         final String applicationId = request.getApplicationId();
         final String applicationName = request.getApplicationName();
-        final String applicationKey = request.getApplicationKey();
-        final String applicationSecret = request.getApplicationSecret();
-        final String masterPublicKey = request.getMasterPublicKey();
+        final String mobileSdkConfig = request.getMobileSdkConfig();
 
-        TestConfigEntity appConfig = getOrCreateTestAppConfig(applicationId);
+        final TestConfigEntity appConfig = getOrCreateTestAppConfig(applicationId);
+
+        final String applicationKey;
+        final String applicationSecret;
+        final String masterPublicKey;
+
+        if (mobileSdkConfig != null) {
+            final SdkConfiguration config;
+            try {
+                config = SdkConfigurationSerializer.deserialize(mobileSdkConfig);
+            } catch (Exception ex) {
+                logger.warn("Invalid mobile SDK configuration", ex);
+                throw new AppConfigInvalidException("Invalid mobile SDK configuration");
```

Comment:
I suggest keeping the cause and logging the whole stack trace only in the handler.
```suggestion
                logger.warn("Invalid mobile SDK configuration: {}", ex.getMessage());
                throw new AppConfigInvalidException("Invalid mobile SDK configuration", ex);
```